### PR TITLE
fix(ui): always-opaque toolbar background, darker primary hover

### PR DESCRIPTION
**Pull request by ajay-fluxon**
## Summary

- **Sticky toolbar bleed-through.** `.test-cases-toolbar` is `position: sticky` but had no background, so test-case rows scrolling underneath showed through it once pinned. Fixed by making the background always opaque (`var(--background)`), with a bottom border, rounded corners (`--radius-lg`, matching the question cards below), and a subtle shadow.
- **Primary button hover.** `.ui-button--primary` / `.ui-icon-button--primary` lightened on hover via `opacity: 0.9`, reading as washed out. They now darken via a new `--primary-hover` token (`#0925b2` light, `#324dd9` dark).
- **Demo spacing.** `#runner-host` in the vanilla demo gets `padding-top: 8px`, matching how the consuming app already wraps the component (`pt: "8px"` in its own layout), so the demo reflects real usage.

## Demo

https://www.loom.com/share/a34716a19ce34061a3e085add72272c2

## Why not track a "stuck" state instead

A prior attempt (#98, closed in favor of this PR) kept the toolbar transparent in normal flow and toggled an opaque background only once pinned, detected via a sentinel element + `IntersectionObserver`/scroll listener. That went through several rounds of genuine correctness bugs — missed crossings on fast/slow scrolls, `overflow-x: hidden` on `body` causing `overflow-y` to compute to `auto` and confusing scroll-root detection, initial-render timing — each fixed but replaced by the next edge case. This PR removes all of that: no JS, no scroll tracking, no sentinel. The toolbar is simply always opaque, which is simpler and can't have the same bug class.

## Test plan

- [x] `npm run lint` — 0 errors (7 pre-existing `no-explicit-any` warnings in untouched files)
- [x] `npx tsc --noEmit -p tsconfig.react.json` — clean
- [x] `npm test` — 229/229 tests pass
- [x] Verified locally against a real consumer app (symlinked), including the `overflow-x: hidden` on `body` case that broke the previous approach
- [ ] Manual: confirm the toolbar's rounded corners and shadow look right against your app's page background

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/components/llm-test-runner/test-cases/test-cases-toolbar.css [hunk 1]
```diff
@@ -10,6 +9,10 @@
   position: sticky;
   top: var(--llmtr-sticky-top, 0px);
   z-index: var(--z-sticky);
+  background: var(--background);
+  border-bottom: var(--border-width) solid var(--border);
+  border-radius: var(--radius-lg);
```
[the remark this answers is not in the document]
Comment (ajay-fluxon) [reply src/components/llm-test-runner/test-cases/test-cases-toolbar.css:14]: its intended

**Review by arup-fluxon — APPROVED** [review]
